### DISCUSSION TITLE
assetpack file and bundle file are duplicated in aab file

### DISCUSSION THREAD

**Discussion by heejeong-park-dev** [jelte/be.khepri.play.assetdelivery.addressables#66] [Q&A]
```CSharp
var buildPath = Path.Combine(Addressables.BuildPath, "Android");
var assetPack = AssetPackBuilder.CreateAssetPacks(textureCompressionFormat, buildPath);
assetPack.SplitBaseModuleAssets = true;
AssetPackConfigSerializer.SaveConfig(assetPack);
AppBundlePublisher.Build(buildPlayerOptions, assetPack, true);
```
I checked the aab file that was built.
![image](https://user-images.githubusercontent.com/107017511/195035452-7a507ad4-c455-45d6-a014-61348f8aedf4.png)
And I checked there is a bundle file in a `base_assets` folder.
![image](https://user-images.githubusercontent.com/107017511/195038018-280ae5ba-2b55-4180-bdef-7e7b102f5018.png)
Also, I checked the assetpack file is included in the aab root directory.

So, I wrote the code like below.
```CSharp
...
AssetPackConfigSerializer.SaveConfig(assetPack);
var bundles = Directory.GetFiles(buildPath, "*.bundle", SearchOption.TopDirectoryOnly);
foreach (var bundle in bundles)
{
    File.Delete(bundle);
}

AppBundlePublisher.Build(buildPlayerOptions, assetPack, true);
```
I got a smaller size of aab file but the game won’t work because it couldn’t find a built-in shader.

And then I fix the code like this
```CSharp
if (!bundle.Contains("unitybuiltinshader"))
    File.Delete(bundle);
```
It’s working now and aab file is smaller then before.

I wonder is there anyway handle built-in shader and minimize aab file without hard coding?

**Comment by szeleneyr** [comment]
I used a similar workaround as described here which works fine:

https://github.com/jelte/be.khepri.play.assetdelivery.addressables/issues/62

**Reply by a-l-e-x-d-s-9** [reply]
@szeleneyr @heejeong-park-dev Can you please help me to understand.
This workaround should be added to the function CreateConfig()? 

Here is the original code:
``` c#
        private const string CreateAction = "Create config for Addressables Groups";
        
        [MenuItem(GoogleEditorMenu.MainMenuName + "/" + CreateAction, false, 101)]
        public static void CreateConfig()
        {
            var assetPackConfig = AssetPackBuilder.CreateAssetPacks(TextureCompressionFormat.Default);
            if (assetPackConfig == null)
            {
                EditorUtility.DisplayDialog(CreateAction, "Unable to create AssetPack config. Make sure Addressables asset bundles have been build!", "Ok");
                return;
            }
        }
```

Following this workaround, I changed the code to:
``` C#
        private const string CreateAction = "Create config for Addressables Groups";

        [ MenuItem( GoogleEditorMenu.MainMenuName + "/" + CreateAction, false, 101 ) ]
        public static void CreateConfig()
        {
            var buildPath = Path.Combine( Addressables.BuildPath, "Android" );
            var assetPack = AssetPackBuilder.CreateAssetPacks( TextureCompressionFormat.Default, buildPath );
            assetPack.SplitBaseModuleAssets = true;
            AssetPackConfigSerializer.SaveConfig( assetPack );

            var bundles = Directory.GetFiles( buildPath, "*.bundle", SearchOption.TopDirectoryOnly );
            foreach ( var bundle in bundles )
            {
                if ( !bundle.Contains( "unitybuiltinshader" ) )
                {
                    File.Delete( bundle );
                }
            }
        }
``` 

Now I need to build the addressables. Then run "Create config for Addressables Groups".
And build with Google > Build Android App Bundle...
Right?
It should create a proper aab, without duplications of addressable bundles, but with all Addressable Groups that were configured to be included inside the aab?
What version of Unity Addressables are you using? 1.18.19 or something else?
What version of Unity are you using? 2020.3.40f1, 2021.3.11f1, something else?
Should it solve #62, possibly #64?
